homeassistant/components/ozw/light.py

```diff
@@ -42,45 +55,64 @@ def byte_to_zwave_brightness(value):
     return 0
 
 
-class ZwaveDimmer(ZWaveDeviceEntity, LightEntity):
-    """Representation of a Z-Wave dimmer."""
+class ZwaveLight(ZWaveDeviceEntity, LightEntity):
+    """Representation of a Z-Wave light."""
 
     def __init__(self, values):
         """Initialize the light."""
         super().__init__(values)
+        self._color_channels = None
+        self._hs = None
+        self._white = None
         self._supported_features = SUPPORT_BRIGHTNESS
         # make sure that supported features is correctly set
         self.on_value_update()
 
     @callback
     def on_value_update(self):
         """Call when the underlying value(s) is added or updated."""
-        self._supported_features = SUPPORT_BRIGHTNESS
         if self.values.dimming_duration is not None:
             self._supported_features |= SUPPORT_TRANSITION
 
+        if self.values.color is None and self.values.color_channels is None:
+            return
+
+        if self.values.color is not None:
+            self._supported_features |= SUPPORT_COLOR
+
+        if self.values.color_channels is not None:
+            self._supported_features |= SUPPORT_WHITE_VALUE
+
+        self._calculate_rgb_values()
+
     @property
     def brightness(self):
         """Return the brightness of this light between 0..255.
 
         Zwave multilevel switches use a range of [0, 99] to control brightness.
         """
-        if "target" in self.values:
```

Comment:
Please inform me why you deleted this code ?
This was explicitly added on request of the OZW team that the target value must be preferred over the regular value. Did it cause some issue we're not aware of ?

Reply:
I think I got caught up in cleaning up the if statements, I'll put them back. Those are for the dimming correct?

Reply:
> I think I got caught up in cleaning up the if statements, I'll put them back. Those are for the dimming correct?

Well, when a device is transitioning, devices that support it will have the value it's transitioning into in the target value. Don't know the exact story in Z-Wave specs but it fixed issues with some Z-Wave dimmers a few months back and we placed this in after some conversation with @Fishwaldo.

Reply:
I've put them back in the latest commit :) Thanks for catching that.

Reply:
On Color - target is the final color value the device is transitioning to. Not a brightness value.